src/redux/addCash.js

```diff
@@ -76,7 +90,99 @@ export const addCashNewPurchaseTransaction = txDetails => (
   savePurchaseTransactions(updatedPurchases, accountAddress, network);
 };
 
-export const addCashGetTransferHash = (
+export const addCashGetOrderStatus = (
+  referenceInfo,
+  destCurrency,
+  orderId,
+  paymentResponse,
+  sourceAmount
+) => async (dispatch, getState) => {
+  logger.log('[add cash] - watch for order status', orderId);
+  const { network } = getState().settings;
+  const getOrderStatus = async (
+    referenceInfo,
+    destCurrency,
+    orderId,
+    paymentResponse,
+    sourceAmount
+  ) => {
+    try {
+      const { data, orderStatus, transferId } = await trackWyreOrder(
+        referenceInfo,
+        orderId,
+        network
+      );
+
+      dispatch({
+        payload: orderStatus,
+        type: ADD_CASH_UPDATE_CURRENT_ORDER_STATUS,
+      });
+
+      const isFailed = orderStatus === WYRE_ORDER_STATUS_TYPES.failed;
+
+      if (isFailed) {
+        logger.sentry('Wyre order data failed', data);
+        captureMessage(
+          `Wyre final check - order status failed - ${referenceInfo.referenceId}`
+        );
+        analytics.track('Purchase failed', {
+          category: 'add cash',
+          error_category: get(data, 'errorCategory', 'unknown'),
+          error_code: get(data, 'errorCode', 'unknown'),
+        });
+      }
+
+      if (transferId) {
+        dispatch({
+          payload: transferId,
+          type: ADD_CASH_UPDATE_CURRENT_TRANSFER_ID,
+        });
+        referenceInfo.transferId = transferId;
+        dispatch(
+          addCashGetTransferHash(referenceInfo, transferId, sourceAmount)
+        );
+        analytics.track('Purchase completed', {
+          category: 'add cash',
+        });
+      } else if (!isFailed) {
+        setTimeout(
+          () =>
+            getOrderStatus(
+              referenceInfo,
+              destCurrency,
+              orderId,
+              paymentResponse,
+              sourceAmount
+            ),
+          1000
+        );
+      }
+    } catch (error) {
+      captureException(error);
+      setTimeout(
```

Comment:
Would this run forever if it keeps throwing the same error?

Reply:
I'll be making a separate PR for limits on places we have loops for requests like this.